$MirOS: ports/www/firesomething/patches/patch-security_coreconf_MidnightBSD_mk,v 1.2 2008/12/17 21:34:48 bsiegert Exp $
--- security/coreconf/MidnightBSD.mk.orig	Wed Dec 17 23:12:55 2008
+++ security/coreconf/MidnightBSD.mk	Wed Dec 17 23:12:57 2008
@@ -20,6 +20,7 @@
 # the Initial Developer. All Rights Reserved.
 #
 # Contributor(s):
+# * The MirOS Project
 #
 # Alternatively, the contents of this file may be used under the terms of
 # either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -37,11 +38,13 @@
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
 
-DEFAULT_COMPILER	= gcc
-CC			= gcc
-CCC			= g++
+CC			?= gcc
+CXX			?= g++
+DEFAULT_COMPILER	= ${CC}
+CCC			= ${CXX}
 RANLIB			= ranlib
 
+#XXX what about sparc64, amd64?
 ifeq ($(OS_TEST),alpha)
 CPU_ARCH		= alpha
 else
@@ -50,8 +53,8 @@ endif
 
 OS_CFLAGS		= $(DSO_CFLAGS) -ansi -Wall -Wno-switch -DFREEBSD -DHAVE_STRERROR -DHAVE_BSD_FLOCK
 
-DSO_CFLAGS		= -fPIC
-DSO_LDOPTS		= -shared -Wl,-soname -Wl,$(notdir $@)
+DSO_CFLAGS		= -fPIC -DPIC
+DSO_LDOPTS		= -shared -fPIC -Wl,-soname,lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX) -Bsymbolic
 
 #
 # The default implementation strategy for FreeBSD is pthreads.
@@ -65,21 +68,11 @@ endif
 
 ARCH			= freebsd
 
-MOZ_OBJFORMAT		:= $(shell test -x /usr/bin/objformat && /usr/bin/objformat || echo elf)
+MOZ_OBJFORMAT		= elf
 
-ifeq ($(MOZ_OBJFORMAT),elf)
 DLL_SUFFIX		= so
-else
-DLL_SUFFIX		= so.1.0
-endif
 
 MKSHLIB			= $(CC) $(DSO_LDOPTS)
-ifdef MAPFILE
-	MKSHLIB += -Wl,--version-script,$(MAPFILE)
-endif
-PROCESS_MAP_FILE = grep -v ';-' $< | \
-        sed -e 's,;+,,' -e 's; DATA ;;' -e 's,;;,,' -e 's,;.*,;,' > $@
 
-G++INCLUDES		= -I/usr/include/g++
-
-INCLUDES		+= -I/usr/X11R6/include
+USE_SYSTEM_ZLIB		= 1
+ZLIB_LIBS		= -lz
